tools/upgrade_versions.py

```diff
@@ -229,13 +229,18 @@ def run(CONF):
         CONF.get('default').get('distro_name'),
         CONF.get('default').get('distro_version'),
         CONF.get('default').get('arch_and_endianness'))
+    commit_updates = CONF.get('default').get('commit_updates')
+    push_updates = CONF.get('default').get('push_updates')
     push_repo_url = CONF.get('default').get('push_repo_url')
     push_repo_branch = CONF.get('default').get('push_repo_branch')
-    committer_name = CONF.get('default').get('committer_name')
-    committer_email = CONF.get('default').get('committer_email')
-
-    REQUIRED_PARAMETERS = ["push_repo_url", "push_repo_branch",
-                           "committer_name", "committer_email"]
+    updater_name = CONF.get('default').get('updater_name')
+    updater_email = CONF.get('default').get('updater_email')
+
+    if push_updates:
+        REQUIRED_PARAMETERS = ["push_repo_url", "push_repo_branch"]
+    else:
+        REQUIRED_PARAMETERS = []
+    REQUIRED_PARAMETERS += ["updater_name", "updater_email"]
```

Comment:
You could set `REQUIRED_PARAMETERS = ["updater_name", "updater_email"]` first and remove the else statement.